src/persistence/SubjectOperationExecutor.ts

```diff
@@ -89,8 +89,8 @@ export class SubjectOperationExecutor {
             !this.updateSubjects.length &&
             !this.removeSubjects.length &&
             !this.relationUpdateSubjects.length &&
-            !subjects.some(subject => !subject.junctionInserts.length) &&
-            !subjects.some(subject => !subject.junctionRemoves.length))
+            !subjects.some(subject => !!subject.junctionInserts.length) &&
+            !subjects.some(subject => !!subject.junctionRemoves.length))
```

Comment:
reversal conditions make it hard to read and understand. maybe better to replace it with `subjects.every(subject => !subject.junctionInserts.length)`. Is it same?